### DISCUSSION TITLE
Manually send reports

### DISCUSSION THREAD

**Discussion by mst-jc** [ACRA/acra#1281] [Q&A]
is there a way to manually send the reports so that ACRA collects the crash logs and when I require it send them?
My use case is as follows:
I have custom sender to be able to send the reports to my server, however in order to actually send data to my server the user must have a valid session (infrastructure requirements not mine, so security pointed that i can only have a session if the user actually opens the app and and logs back in) so if the app crashes while the user has no current internet but has a valid session I want it to collect the crash but that only send it when he decides to open the app again and log back in.

**Comment by F43nd1r** [comment]
You'll have to write some code yourself, but ACRA should provide enough extension points to enable this.
Have a look at the advanced scheduler as a starting point https://github.com/ACRA/acra/blob/master/acra-advanced-scheduler/src/main/java/org/acra/scheduler/AdvancedSenderScheduler.kt
